### PR TITLE
One runner for the entire Funding Service

### PR DESCRIPTION
**Pull request by samuelhwilliams**
This patch pulls in the "post award" services and combines everything into a single docker-compose file that can run the entire Funding Service. Optionally, you can just run the pre- or -post- award services if that's all you need at the time.

This makes a few refactors:

* Removes the concept of customisable workspaces. Previously the two scripts took a "workspace directory" letting the user say where to checkout repos. However, the docker-compose explicitly required these to be checked out in the parent directory, so it wasn't actually customisable. This patch now expects all repos inside an `apps` directory within this repo, to keep everything contained.
* Pre-award services used to be accessed over HTTP on `localhost`. They now match post-award and must be accessed over HTTPS on defined `.levellingup.gov.localhost` domains. This increases parity with prod and allows Authenticator to be used correctly for logging in/out via Azure AD.
* Pre-award's `fund_store` now automatically seeds with funding data on startup, which matches what post-award's `data-store` does. This should ease the developer setup.
* Git cloning is now done over SSH, not HTTPS.

### Future possibilities
* Could re-introduce the idea of workspace directories if developers are very keen. I'm descoping that for now, as I think it's not a hard requirement and I'm not immediately sure how docker-compose would be edited to handle it.
* There are a few 'setup' scripts - `reset-all-repos.sh` and `install-venv-all-repos.sh`. Possibly these could be combined into a single script, tidied up, given more obvious flags and some better help docs.
* Clean up/standardise environment variable handling for docker-compose, and env var naming across pre and post award.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,65 +1,99 @@
...
+
+To run just the post-award services, execute `make post up`.
+
+
+### Service domains:
+
+* Authenticator: https://one-authenticator.levellingup.gov.localhost:3004
+  * Example URL: https://one-authenticator.levellingup.gov.localhost:3004/service/magic-links/new?fund=cof&round=r2w3
+
+* Apply: https://one-frontend.levellingup.gov.localhost:3008
```
Comment (nuwan-samarasinghe) [inline README.md:45]: Why do we have one prefix ?

File: docker-localstack/setup-awslocal.sh [hunk 2]
```diff
@@ -99,3 +100,15 @@ else
 
   echo "Created SQS Queue: $SQS_NOTIFICATION_QUEUE_URL!"
 fi
+# <- Pre-award environment ->
+
+
+# <- Post-award environment ->
+for bucket in ${AWS_S3_BUCKET_FAILED_FILES} ${AWS_S3_BUCKET_SUCCESSFUL_FILES} ${AWS_S3_BUCKET_FIND_DOWNLOAD_FILES}; do
```
Comment (nuwan-samarasinghe) [inline docker-localstack/setup-awslocal.sh:107]: Cant we generalise following to create the list of S3 buckets

`
if awslocal s3 ls | grep -q $AWS_BUCKET_NAME; then
  echo "Bucket already exists!"
else
  awslocal s3api \
  create-bucket --bucket $AWS_BUCKET_NAME \
  --create-bucket-configuration LocationConstraint=$AWS_REGION \
  --region $AWS_REGION
  echo "Created Bucket $AWS_BUCKET_NAME!"
  awslocal s3api \
  put-bucket-cors --bucket $AWS_BUCKET_NAME \
  --cors-configuration '{
    "CORSRules": [
      {
        "AllowedOrigins": ["*"],
        "AllowedMethods": ["GET", "PUT", "POST", "DELETE", "HEAD"],
        "AllowedHeaders": ["*"]
      }
    ]
  }'
  echo "Created CORS rule!"
fi

if awslocal s3 ls | grep -q $AWS_MSG_BUCKET_NAME; then
  echo "Bucket already exists!"
else
  awslocal s3api \
  create-bucket --bucket $AWS_MSG_BUCKET_NAME \
  --create-bucket-configuration LocationConstraint=$AWS_REGION \
  --region $AWS_REGION
  echo "Created Bucket $AWS_BUCKET_NAME!"
  awslocal s3api \
  put-bucket-cors --bucket $AWS_MSG_BUCKET_NAME \
  --cors-configuration '{
    "CORSRules": [
      {
        "AllowedOrigins": ["*"],
        "AllowedMethods": ["GET", "PUT", "POST", "DELETE", "HEAD"],
        "AllowedHeaders": ["*"]
      }
    ]
  }'
  echo "Created CORS rule!"
fi
`

**Review by nuwan-samarasinghe — APPROVED** [review]
Nice work

**Review by NarenderRajuB — APPROVED** [review]
LGTM

File: docker-localstack/setup-awslocal.sh (hunk 2 above)
Comment (samuelhwilliams) [reply docker-localstack/setup-awslocal.sh:90]: Patched in 1e1c7aabbff8b5d8eb161991b319f9991b19acb8

File: .env.example [hunk 3]
```diff
@@ -1,11 +1,25 @@
-# COPY TO .env then see README for details on how to obtain the values
-GOV_NOTIFY_API_KEY=
+# <- Pre-award environment ->
+GOV_NOTIFY_API_KEY=<INSERT-GOVUK-NOTIFY-API-KEY-FOR-ACCESS-FUNDING-SERVICE-IF-USING-PRE-AWARD>
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply .env.example:2]: agree, that sounds like a good idea 👍  let's do that in a follow-up

File: docker-compose.yml [hunk 4]
```diff
@@ -158,53 +153,176 @@ services:
...
+    env_file: .awslocal.env
+    environment:
+      - GOV_NOTIFY_API_KEY=${GOV_NOTIFY_API_KEY:?err}
+      - FLASK_DEBUG=1
+      - FLASK_ENV=development
+      - AWS_MSG_BUCKET_NAME=fsd-notification-bucket
+    profiles: [ pre ]
+
+  data-store:
+    # http://github.com/communitiesuk/funding-service-design-post-award-data-store
```
Comment (MarcUsher) [inline docker-compose.yml:221]: V minor point but this feels like a handy thing to link to the GitHub repo where the service is declared in the docker-compose, could be nice to add it for the rest of them?

Comment (samuelhwilliams) [reply docker-compose.yml:221]: yep yep, will sneak this in as a followup PR, good shout

**Pull request merged**